include/wimlib/compiler.h

```diff
@@ -142,21 +157,127 @@
 #endif
 
 /* STATIC_ASSERT() - verify the truth of an expression at compilation time.  */
+#ifdef _MSC_VER
+#define STATIC_ASSERT(expr) assert(expr)
```

Comment:
Current Visual Studio (even without `/std:clatest`) defines an `__STDC_VERSION__` that is higher `201112L` and it has support for `_Static_assert()`, so this redef can be avoided (especially as you really do want static asserts to work when introducing support for a new compiler).